oxide/lib.go

```diff
@@ -111,47 +86,92 @@ func parseBaseURL(baseURL string) (string, error) {
 	return b, nil
 }
 
-// WithToken overrides the token used for authentication.
-func (c *Client) WithToken(token string) {
-	c.token = token
+// Opt is a configuration option to initialize a Client.
+type Opt func(*Client) error
+
+// FromEnv initializes the Client using environment variables.
+func FromEnv() Opt {
+	return func(c *Client) error {
+		token := os.Getenv(TokenEnvVar)
+		if token == "" {
+			return fmt.Errorf("the environment variable %s must be set with your API token", TokenEnvVar)
+		}
+		c.token = token
+
+		host := os.Getenv(HostEnvVar)
+		if host == "" {
+			return fmt.Errorf("the environment variable %s must be set with the host of the Oxide API", HostEnvVar)
+		}
+		baseURL, err := parseBaseURL(host)
+		if err != nil {
+			return err
+		}
+		c.server = baseURL
+
+		return nil
+	}
 }
 
-type userAgentTransport struct {
-	userAgent string
-	base      http.RoundTripper
-	client    *Client
+// WithHost overrides the client server with the specified one.
+func WithHost(host string) Opt {
+	return func(c *Client) error {
+		if host != "" {
+			baseURL, err := parseBaseURL(host)
+			if err != nil {
+				return err
+			}
+
+			c.server = baseURL
+		}
+		return nil
+	}
 }
 
-func (t userAgentTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	if t.base == nil {
-		return nil, errors.New("RoundTrip: no Transport specified")
+// WithToken overrides the client token with the specified one.
+func WithToken(token string) Opt {
+	return func(c *Client) error {
+		if token != "" {
+			c.token = token
+		}
+		return nil
 	}
+}
```

Comment:
We generally want our options to be optional settings. `FromEnv`, `WithHost` and `WithToken`  don't really fit that category entirely as you must have the host and token set one way or another. 

Let's remove these three and continue to take them as parameters.

Reply:
I went in a different direction here that uses a new `Config` type to configure the `Client`. Let me know what you think? I felt it was cleaner for the use case where we want to use environment variables to configure the client.